lib/libexfat.c

```diff
@@ -600,20 +601,22 @@ int exfat_set_volume_serial(struct exfat_blk_dev *bd,
 	int ret;
 	struct pbr *ppbr;
 
-	ppbr = malloc(bd->sector_size);
+	ppbr = malloc(EXFAT_MAX_SECTOR_SIZE);
 	if (!ppbr) {
 		exfat_err("Cannot allocate pbr: out of memory\n");
 		return -1;
 	}
 
 	/* read main boot sector */
-	ret = exfat_read_sector(bd, (char *)ppbr, BOOT_SEC_IDX);
+	ret = exfat_read(bd->dev_fd, (char *)ppbr, EXFAT_MAX_SECTOR_SIZE,
+			BOOT_SEC_IDX);
 	if (ret < 0) {
 		exfat_err("main boot sector read failed\n");
 		ret = -1;
 		goto free_ppbr;
 	}
 
+	bd->sector_size = 1 << ppbr->bsx.sect_size_bits;
```

Comment:
Callers have high chance not to known that this function change device's sector size. I think we'd rather change the sector size in main() than in this function.

Reply:
Why ? What problem ? Can you explain what is problem ?

Reply:
this function is used for tune.exfat and exfatlabel. if we move bd->sector_size set to main(), we should add it to each main of  label and tune. It will make duplicate code and looks strange.

Reply:
Okay